### PR TITLE
[NETBEANS-5244] Add the project settings for XML configuration of phpDocumentor

### PR DESCRIPTION
**Pull request by junichi11**
https://issues.apache.org/jira/browse/NETBEANS-5244

- Add settings for an XML configuration file
- Add parameters (`--config /path/to/configuration/file.xml`)

netbeans-5244-phpDocumentor-configuration-path

### CODE REVIEW

**Comment by junichi11** [comment]
@tmysik Sorry for the many review requests. Don't worry, I'll wait anyway :) Thanks.

**Review by KacerCZ — APPROVED** [review]
Tested, works.

File: php/php.phpdoc/src/org/netbeans/modules/php/phpdoc/PhpDocScript.java [hunk 1]
```diff
@@ -145,13 +149,24 @@ private String sanitizePath(String path) {
...
     }
 
-    private List<String> getParameters(String sanitizedPhpDocTarget, PhpModule phpModule) {
+    private List<String> getAllParameters(String sanitizedPhpDocTarget, PhpModule phpModule) {
+        List<String> params = new ArrayList<>(getDefaultParameters(sanitizedPhpDocTarget, phpModule));
+        if (PhpDocPreferences.isConfigurationEnabled(phpModule)) {
+            String configurationPath = PhpDocPreferences.getPhpDocConfigurationPath(phpModule);
+            if (!StringUtils.isEmpty(configurationPath)) {
+                params.add(PARAM_CONFIG);
+                params.add(sanitizePath(PhpDocPreferences.getPhpDocConfigurationPath(phpModule)));
```
Comment (tmysik) [inline php/php.phpdoc/src/org/netbeans/modules/php/phpdoc/PhpDocScript.java:158]: If the path is incorrect, PhpDoc itself will complain, I guess, so we don't need to run the validator here. Or do we? :thinking: 

AFAIR we validate project properties before each run and if they are incorrect, the Project Properties dialog is opened.

@junichi11 @KacerCZ What do you think, guys?

**Review by tmysik — APPROVED** [review]

**Comment by tmysik** [comment]
@junichi11 Just added one comment. Feel free to merge it as it is now, it is more a question. Thank you!

File: php/php.phpdoc/src/org/netbeans/modules/php/phpdoc/PhpDocScript.java (hunk 1 above)
Comment (KacerCZ) [reply php/php.phpdoc/src/org/netbeans/modules/php/phpdoc/PhpDocScript.java:158]: Properties are opened only if target directory for generated documentation is not valid.
`sanitizePath()` only replaces backslashes on Windows with slashes.

Comment (tmysik) [reply php/php.phpdoc/src/org/netbeans/modules/php/phpdoc/PhpDocScript.java:158]: > Properties are opened only if target directory for generated documentation is not valid.

So, should not it be opened also if the enabled configuration file is invalid, typically it does not exist?

Comment (junichi11) [reply php/php.phpdoc/src/org/netbeans/modules/php/phpdoc/PhpDocScript.java:158]: I'll improve it.

Comment (tmysik) [reply php/php.phpdoc/src/org/netbeans/modules/php/phpdoc/PhpDocScript.java:158]: Thank you.

Comment (junichi11) [reply php/php.phpdoc/src/org/netbeans/modules/php/phpdoc/PhpDocScript.java:207]: Fixed.
netbeans-5244-phpdocumentor-invalid-settings

Comment (tmysik) [reply php/php.phpdoc/src/org/netbeans/modules/php/phpdoc/PhpDocScript.java:207]: Nicely done, thank you.

**Comment by junichi11** [comment]
Thank you for your reviews.

**Pull request merged**